lib/hmmlearn/hmm.py

```diff
@@ -955,11 +943,38 @@ def _accumulate_sufficient_statistics(self, stats, X, lattice,
 
         with np.errstate(under="ignore"):
             post_comp_mix = post_comp[:, :, None] * post_mix
-        stats['post_comp_mix'].append(post_comp_mix)
 
         stats['post_mix_sum'] += post_comp_mix.sum(axis=0)
         stats['post_sum'] += post_comp.sum(axis=0)
 
+        if 'm' in self.params:  # means stats
+            stats['m_n'] += np.einsum('ijk,il->jkl', post_comp_mix, X)
+
+        if 'c' in self.params:  # covariance stats
+            centered = X[:, None, None, :] - self.means_
+
+            def outer_f(x):  # Outer product over features.
+                return x[..., :, None] * x[..., None, :]
+
+            def norm_last(x):  # Much faster than (x**2).sum(-1).
+                return np.einsum('...i,...i', x, x)
```

Comment:
minor nit-pick:  the definitions of these two functions are now duplicated in `_accumulate_sufficient_statistics` and also in `_do_mstep`.  perhaps it'd be better to define them once in global scope of this file (perhaps renamed as `_outer_f` and `_norm_last`,  at the cost of making the calculations slightly less locally readable

Reply:
I think these are short enough that it's actually OK and more readable to duplicate the implementations to have them readily available (in fact norm_last can even just be inlined...)